domain/secretbackend/service/modelsecretbackendservice.go

```diff
@@ -81,8 +85,45 @@ func (s *ModelSecretBackendService) SetModelSecretBackend(ctx context.Context, b
 
 		}
 	}
+
+	if err := s.checkBackendCompatibility(ctx, backendName); err != nil {
+		return errors.Errorf("checking backend compatibility for %q: %w", backendName, err)
+	}
+
 	if err := s.st.SetModelSecretBackend(ctx, s.modelID, backendName); err != nil {
 		return errors.Errorf("setting model secret backend for %q: %w", s.modelID, err)
 	}
 	return nil
 }
+
+// checkBackendCompatibility verifies if the specified secret backend is
+// compatible with the current Juju version.
```

Comment:
I would look to re-phrase this wording. It isn't the current Juju version as this is implying. It is checking backend compatibility with the model's target Juju version.